src/Microsoft.DncEng.PatGenerator/AzureDevOpsPATScopes.cs

```diff
@@ -0,0 +1,157 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.DncEng.PatGenerator
+{
+    [AttributeUsage(AttributeTargets.Field)]
+    public class ScopeShortHandAttribute : Attribute
+    {
+        public string Category { get; }
+        public string Permissions { get; }
+
+        public ScopeShortHandAttribute(string category, string shortName)
+        {
+            Category = category;
+            Permissions = shortName;
+        }
+    }
+
+    /// <summary>
+    /// Available Azure DevOps Scopes
+    /// See <see cref="https://docs.microsoft.com/en-us/azure/devops/integrate/get-started/authentication/oauth?view=azure-devops#scopes"/>
+    /// for more information.
+    /// 
+    /// A flags enumeration is used to coalesce different permissions levels. For example,
+    /// if 'build' and 'build_execute' were supplied together, 'build' is not useful since it's
+    /// implied in build_execute.
+    /// </summary>
+    [Flags]
+    public enum AzureDevOpsPATScopes
+    {
+        // Build - Bits 0x3
+
+        [ScopeShortHand("build", "r")]
+        build = 0x1,
+        [ScopeShortHand("build", "re")]
+        build_execute = 0x2 | build,
+
+        // Code - Bits 0x3C
+
+        [ScopeShortHand("code", "r")]
+        code = 0x4,
+        [ScopeShortHand("code", "s")]
+        code_status = 0x8,
+        [ScopeShortHand("code", "rw")]
+        code_write = 0x10 | code | code_status,
+        [ScopeShortHand("code", "m")]
+        code_manage = 0x20 | code_write,
+
+        // Packaging - Bits 0x1C0
+
+        [ScopeShortHand("package", "r")]
+        packaging = 0x40,
+        [ScopeShortHand("package", "rw")]
+        packaging_write = 0x80 | packaging,
+        [ScopeShortHand("package", "m")]
+        packaging_manage = 0x100 | packaging_write,
+
+        // Symbols - Bits 0xE00
+
+        [ScopeShortHand("symbols", "r")]
+        symbols = 0x200,
+        [ScopeShortHand("symbols", "rw")]
+        symbols_write = 0x400 | symbols,
+        [ScopeShortHand("symbols", "m")]
+        symbols_manage = 0x800 | symbols_write,
```

Comment:
Do we want to add scopes for releases as well?

Reply:
Added. Also added some others, but not all. Help output altered to look like:

```
  --scopes <scopes> (REQUIRED)                PAT scopes. Valid values are: 'build' - Build (read)
                                              'build_execute' - Build (read and execute)
                                              'code' - Code (read)
                                              'code_status' - Code (update commit status)
                                              'code_write' - Code (read and write)
                                              'code_manage' - Code (read, write, and manage)
                                              'packaging' - Packaging (read)
                                              'packaging_write' - Packaging (read and write)
                                              'packaging_manage' - Packaging (read, write, and manage)
                                              'symbols' - Symbols (read)
                                              'symbols_write' - Symbols (read and write)
                                              'symbols_manage' - Symbols (read, write and manage)
                                              'release' - Release (read)
                                              'release_execute' - Release (read, write and execute)
                                              'release_manage' - Release (read, write, execute and manage)
                                              'profile' - User profile (read)
                                              'profile_write' - User profile (write)
                                              'variablegroups_read' - Variable Groups (read)
                                              'variablegroups_write' - Variable Groups (read, create)
                                              'variablegroups_manage' - Variable Groups (read, create and manage)
                                              'work' - Work items (read)
                                              'work_write' - Work items (read and write)
                                              'work_full' - Work items (full).
```